honeybee/logutil.py

```diff
@@ -1,29 +1,42 @@
 import logging
 from logging.handlers import TimedRotatingFileHandler
+import os
 
 
-def get_logger(name):
+def _get_log_folder():
+    home_folder = os.getenv('HOME') or os.path.expanduser('~')
+    log_folder = os.path.join(home_folder, '.honeybee')
+    if not os.path.isdir(log_folder):
+        os.mkdir(log_folder)
+    return log_folder
+
+
+def get_logger(name, filename='honeybee.log'):
     """Get a logger to be used for each module.
 
-    Use this only once in __init__.py file.
+    Args:
+        name: Logger name. The good practice is to set it to __init__ from inside each
+            modules.
+        filename: Logger filename (Default: honeybee.log).
     """
     logger = logging.getLogger(name)
     logger.setLevel(logging.DEBUG)
 
 
     # create a file handler to log debug and higher level logs
-    _file_handler = TimedRotatingFileHandler('honeybee.log', when='midnight')
-    _file_format = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
-    _file_handler.setFormatter(_file_format)
-    _file_handler.setLevel(logging.DEBUG)
+    log_file = os.path.join(_get_log_folder(), filename)
+    file_handler = TimedRotatingFileHandler(log_file, when='midnight')
+    file_format = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    file_handler.setFormatter(file_format)
+    file_handler.setLevel(logging.DEBUG)
 
     # create a console handler that only prints out errors and warnings
-    _stream_handler = logging.StreamHandler()
-    _stream_format = logging.Formatter('%(name)s - %(levelname)s - %(message)s')
-    _stream_handler.setFormatter(_stream_format)
-    _stream_handler.setLevel(logging.WARNING)
+    stream_handler = logging.StreamHandler()
+    stream_format = logging.Formatter('%(name)s - %(levelname)s - %(message)s')
+    stream_handler.setFormatter(stream_format)
+    stream_handler.setLevel(logging.WARNING)
 
-    logger.addHandler(_file_handler)
-    logger.addHandler(_stream_handler)
+    logger.addHandler(file_handler)
```

Comment:
Why do we want to send this to a file by default? Would an output in the terminal not suffice? And then one can give the option of stream to a file?

Reply:
It is very easy to miss the error messages in terminal as we have been experiencing with the plugins right now. Specially with our users who don't know how to code asking them to add a stream to file will be a hard task. I prefer to have them in a file and ask people to look into the file or share the file for debugging.

Reply:
That makes sense. In that case can we set it to debug by default but give the option of setting it to something else? I'm thinking of the pollination stuff which most likely won't need the DEBUG level by default.

Reply:
Oh OK. I was not clear. The default settings is mainly for developers to be able to debug the code when it fails for non-developer users! 😄 By users here I meant the Grasshopper / Dynamo users who would send this to us for debugging their failed runs. And files are much better than screen-capture in such cases.